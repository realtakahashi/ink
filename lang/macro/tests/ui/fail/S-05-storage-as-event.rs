use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod storage_as_event {
    #[ink(storage)]
    #[ink(event)] // We cannot have #[ink(event)] if we already have #[ink(storage)]
    struct StorageAsEvent {}

    impl StorageAsEvent {
        #[ink(constructor)]
        fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        fn message(&self) {}
    }
}

fn main() {}
