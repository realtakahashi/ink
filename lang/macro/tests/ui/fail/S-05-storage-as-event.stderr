error: invalid ink! attribute found for `#[ink(storage)]` struct
 --> $DIR/S-05-storage-as-event.rs:6:10
  |
6 |     #[ink(event)] // We cannot have #[ink(event)] if we already have #[ink(storage)]
  |          ^^^^^^^
